Polyhedron/demo/Polyhedron/triangulate_primitive.h

```diff
@@ -82,25 +83,29 @@ class FacetTriangulator
     idPoints.push_back(idPoint);
 
    }
-   triangulate_with_points(idPoints,more_points, normal, item_diag);
+   if(!triangulate_with_points(idPoints,more_points, normal, item_diag))
+     std::cerr<<"Facet not displayed"<<std::endl;
   }
 
   FacetTriangulator(std::vector<PointAndId > &idPoints,
                   const Vector& normal,
                   const double item_diag)
   {
-    triangulate(idPoints, normal, item_diag);
+    if(!triangulate(idPoints, normal, item_diag))
+      std::cerr<<"Facet not displayed"<<std::endl;
   }
   FacetTriangulator(std::vector<PointAndId > &idPoints,
                     const std::vector<typename Kernel::Point_3>& more_points,
                     const Vector& normal,
                     const double item_diag)
   {
-   triangulate_with_points(idPoints, more_points, normal, item_diag);
+   if(!triangulate_with_points(idPoints, more_points, normal, item_diag))
+     std::cerr<<"Facet not displayed"<<std::endl;
+
   }
 
 private:
-  void triangulate( std::vector<PointAndId > &idPoints,
+  bool triangulate( std::vector<PointAndId > &idPoints,
```

Comment:
And what about the memory ownership? Which object is responsible for that memory allocation and should deallocate it?

You should use `std::shared_ptr` and `std::make_shared`. In the Polyhedron demo, C++11 is allowed, and you should no longer call `new` and `delete` anymore.